sdk/core/azure-core/azure/core/serialization.py

```diff
@@ -105,8 +105,11 @@ def default(self, o):  # pylint: disable=too-many-return-statements
                 if hasattr(o, "year") and hasattr(o, "hour"):
                     # astimezone() fails for naive times in Python 2.7, so make make sure o is aware (tzinfo is set)
                     if not o.tzinfo:
-                        return o.replace(tzinfo=TZ_UTC).isoformat()
-                    return o.astimezone(TZ_UTC).isoformat()
+                        iso_formatted = o.replace(tzinfo=TZ_UTC).isoformat()
+                    else:
+                        iso_formatted = o.astimezone(TZ_UTC).isoformat()
+                    # Replace the trailing "+00:00" UTC offset with "Z" (RFC 3339: https://www.ietf.org/rfc/rfc3339.txt)
+                    return iso_formatted.replace("+00:00", "Z")
```

Comment:
this is I think tangential to this specific PR, but I was looking into the [msrest rfc serialization](https://github.com/Azure/msrest-for-python/blob/master/msrest/serialization.py#L1083) and we don't currently support that. Is this something I want to add? 

msrest serialize rfc output:
```python
serialize_rfc(isodate.parse_datetime("0001-01-01T00:00:00Z")) == 'Mon, 01 Jan 0001 00:00:00 GMT'
```

azure.core.serialization
```python
json.dumps(isodate.parse_datetime("0001-01-01T00:00:00Z"), cls=AzureJSONEncoder) == "0001-01-01T00:00:00Z"
```

cc @lmazuel

Reply:
Just to clear, your question has indeed mothing to do with @mccoyp 's PR  :p . But yes, we could think about having a RFC HTTP header format serializer (not same RFC than here)

Reply:
@mccoyp i'm totally wrong here please ignore me